splice_machine/src/test/java/com/splicemachine/derby/test/framework/SpliceUnitTest.java

```diff
@@ -238,6 +241,12 @@ protected void rowContainsQuery(int[] levels, String query,SpliceWatcher methodW
         }
     }
 
+    protected void rowContainsQuery(int[] levels, String query, String costModelVersion, SpliceWatcher methodWatcher, String... contains) throws Exception {
+        methodWatcher.execute(format("set session_property costModel='%s'", costModelVersion));
```

Comment:
moved this method to the single place it is used in `HashableJoinIT` since it is very custom to the test logic itself, we'll think of factoring it out later in `SpliceUnitTest` as you suggest if that comes necessary.

Reply:
I tried to do this in connection builder but then I realized that ITs in the same class are mostly using the same connection for testing. Some tests need multiple connections to test some concurrent behavior. If we do this in connection builder, we need to double the number of connections for each case (1 -> 2, 4 -> 8, etc.) for testing v1 and v2.

I also find the current way more flexible in constructing test. We can easily control when and where we want to test under which cost model. Anyway, we can discuss this further.